src/Orleans.Core.Abstractions/IDs/GrainId.cs

```diff
@@ -14,39 +13,42 @@ namespace Orleans.Runtime
     [Immutable]
     [Serializable]
     [JsonConverter(typeof(GrainIdJsonConverter))]
-    [StructLayout(LayoutKind.Auto)]
     [GenerateSerializer]
     public readonly struct GrainId : IEquatable<GrainId>, IComparable<GrainId>, ISerializable, ISpanFormattable
     {
+        [Id(1)]
+        private readonly GrainType _type;
+
+        [Id(2)]
+        private readonly IdSpan _key;
```

Comment:
Any particular reason for this instead of an auto-prop? I wasn't aware of any JIT quality differences

Reply:
Accessing these struct fields through properties causes the C# compiler to add local copies for each property call, which the JIT is unable to entirely eliminate (and it also reduces inline chances). This is especially noticeable for chained property accesses and for calls that are not inlined (which force entire structs to be copied on stack).